### PR TITLE
Improve increment/decrement with leading zeros in vim mode

### LINKED ISSUE #18360: vim: increment with left padded number
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When incrementing with `<C-a>` or `<C-x>` vim respect the left 0-padding of the number 

`0000ˇ5` with `<C-a>` should display `00006` but Zed may display: `6`

### Environment

```
Zed: v0.154.1 (Zed)
OS: macOS 14.5.0
Memory: 32 GiB
Architecture: aarch64
```

### If applicable, add mockups / screenshots to help explain present your vision of the feature

_No response_

### If applicable, attach your Zed.log file to this issue.

_No response_

### PR DESCRIPTION
**Pull request by ThomAub**
- Closes: #18360

Release Notes:

- Added support for incrementing and decrementing numbers with leading zeros

### CODE REVIEW

**Comment by ThomAub** [comment]
@cla-bot check

**Comment by ThomAub** [comment]
@cla-bot check

File: crates/vim/src/normal/increment.rs [hunk 1]
```diff
@@ -120,10 +123,12 @@ fn increment_decimal_string(mut num: &str, mut delta: i64) -> String {
         u64::MAX
     };
 
-    if result == 0 || !negative {
-        format!("{}", result)
+    if result == 0 {
+        '0'.to_string()
```
Comment (ConradIrwin) [inline crates/vim/src/normal/increment.rs:127]: @ThomAub I think this should still be padded to width?

File: crates/vim/src/normal/increment.rs [hunk 2]
```diff
@@ -213,8 +218,8 @@ fn find_number(
                 begin = Some(offset);
             }
             num.push(ch);
-            println!("pushing {}", ch);
-            println!();
+            // println!("pushing {}", ch); // why do we println here ?
+            // println!();
```
Comment (ConradIrwin) [inline crates/vim/src/normal/increment.rs:222]: oops, this should be deleted, thanks!

File: crates/vim/src/normal/increment.rs (hunk 1 above)
Comment (ThomAub) [reply crates/vim/src/normal/increment.rs:127]: After a double check in vim/neovim 
`1ˇ2` `12 <C-x>`  becomes `0` and stay like this 
One other broken edge case is 
`09ˇ9` `<C-a> <C-x>` is `99` with vim not `099`

File: crates/vim/src/normal/increment.rs (hunk 2 above)
Comment (ThomAub) [reply crates/vim/src/normal/increment.rs:222]: will remove !

File: crates/vim/src/normal/increment.rs (hunk 1 above)
Comment (ConradIrwin) [reply crates/vim/src/normal/increment.rs:127]: Hmm; but `012` becomes `000`.

I think we need to look at the first digit to decide whether to preserve the width? e.g. 100 ctrl-x should become 99 (not 099).

Comment (ThomAub) [reply crates/vim/src/normal/increment.rs:127]: Indeed, I mostly use `<C-a>` to reflow some migrations number and never really looked at all the possibilities of 
`<C-a> <C-x>`.
I will study how [neovim does this](https://github.com/neovim/neovim/blob/64847fbdc908bf0a301b8f1e1814ff71bd425bae/src/nvim/ops.c#L4510) and more tests case in following commits

**Comment by ConradIrwin** [comment]
Thanks! Sorry for not hitting merge before disappearing for two weeks.

**Pull request merged**